src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/validation/FlowValidator.java

```diff
@@ -110,11 +110,7 @@ public void validate(Flow flow, RequestedFlow requestedFlow, Set<String> bulkUpd
 
     private void validateFlowLoop(Flow flow, RequestedFlow requestedFlow) throws InvalidFlowException {
         if (requestedFlow.getLoopSwitchId() != null) {
-            //todo: fix loops for q-in-q and singe switch loops
-            if (requestedFlow.getSrcInnerVlan() != 0 || requestedFlow.getDestInnerVlan() != 0) {
-                throw new InvalidFlowException("Loop for Q-in-Q flows is not implemented",
-                        ErrorType.NOT_IMPLEMENTED);
-            }
+            //todo: fix loops singe switch loops
```

Comment:
you will get conflict here with patch about single switch loops

Reply:
Rebased patch for single-switch flows on this one